src/collectors/windows.plugin/perflib-mssql.c

```diff
@@ -0,0 +1,779 @@
+
+            rrdlabels_add(p->st_access_method_page_splits->rrdlabels, "mssql_instance", p->instanceID, RRDLABEL_SRC_AUTO);
+        }
+
+        rrddim_set_by_pointer(p->st_access_method_page_splits,
+                              p->rd_access_method_page_splits,
+                              (collected_number)p->MSSQLAccessMethodPageSplits.current.Data);
+        rrdset_done(p->st_access_method_page_splits);
+    }
+}
+
+static void do_mssql_errors(PERF_DATA_BLOCK *pDataBlock, struct mssql_instance *p, int update_every)
+{
+    char id[RRD_ID_LENGTH_MAX + 1];
+    PERF_OBJECT_TYPE *pObjectType = perflibFindObjectTypeByName(pDataBlock, p->objectName[NETDATA_MSSQL_SQL_ERRORS]);
+    if (!pObjectType)
+        return;
+
+    if (perflibGetObjectCounter(pDataBlock, pObjectType, &p->MSSQLSQLErrorsTotal)) {
+        snprintfz(id, RRD_ID_LENGTH_MAX, "instance_%s_sql_errors_total", p->instanceID);
+        if (!p->st_sql_errors) {
+            p->st_sql_errors = rrdset_create_localhost("mssql"
+                                                       , id, NULL
+                                                       , "Errors"
+                                                       , "mssql.instance_sql_errors"
+                                                       , "Errors"
+                                                       , "errors/s"
+                                                       , PLUGIN_WINDOWS_NAME
+                                                       , "PerflibMSSQL"
+                                                       , PRIO_MSSQL_SQL_ERRORS
+                                                       , update_every
+                                                       , RRDSET_TYPE_LINE
+                                                       );
+
+            snprintfz(id, RRD_ID_LENGTH_MAX, "mssql_instance_%s_sql_errors_total", p->instanceID);
+            p->rd_sql_errors  = rrddim_add(p->st_sql_errors,
+                                          id,
+                                              "errors",
+                                          1,
+                                          1,
+                                          RRD_ALGORITHM_INCREMENTAL);
+
+            rrdlabels_add(p->st_sql_errors->rrdlabels, "mssql_instance", p->instanceID, RRDLABEL_SRC_AUTO);
+        }
+
+        rrddim_set_by_pointer(p->st_sql_errors,
+                              p->rd_sql_errors,
+                              (collected_number)p->MSSQLAccessMethodPageSplits.current.Data);
+        rrdset_done(p->st_sql_errors);
+    }
+}
+
+int dict_mssql_charts_cb(const DICTIONARY_ITEM *item __maybe_unused, void *value, void *data __maybe_unused) {
+    struct mssql_instance *p = value;
+    int *update_every = data;
+
+    static void (*doMSSQL[])(PERF_DATA_BLOCK *, struct mssql_instance *, int) = {
+        do_mssql_general_stats,
+        do_mssql_errors,
+        NULL, NULL, NULL,
```

Comment:
Is there a reason for the NULL function pointers here?

Reply:
Yes, this is the first PR bringing mssql metrics, the next will fill these null with the functions.

I broke the PR in two different groups, because the second group has the instances and also databases, and thanks this the logic is slight different.